webapp/channels/src/components/user_settings_modal/advanced_display_settings/utils.tsx

```diff
@@ -0,0 +1,307 @@
+    if (sendOnCtrlEnter === 'true' && codeBlockOnCtrlEnter === 'false') {
+        return AdvanceSettings.SEND_ON_CTRL_ENTER;
+    } else if (sendOnCtrlEnter === 'false' && codeBlockOnCtrlEnter === 'true') {
+        return AdvanceSettings.CODE_BLOCK_CTRL_ENTER;
+    }
+    return OnAndOff.OFF;
+};
+
+export const getCtrlSendPreferenceValue = (value: string): {[AdvanceSettings.SEND_ON_CTRL_ENTER]: string;[AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: string} => {
+    if (value === AdvanceSettings.SEND_ON_CTRL_ENTER) {
+        return {[AdvanceSettings.SEND_ON_CTRL_ENTER]: 'true', [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: 'false'};
+    } else if (value === AdvanceSettings.CODE_BLOCK_CTRL_ENTER) {
+        return {[AdvanceSettings.SEND_ON_CTRL_ENTER]: 'false', [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: 'true'};
+    }
+    return {[AdvanceSettings.SEND_ON_CTRL_ENTER]: 'false', [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: 'false'};
+};
+
+export const ctrlSendInputFieldData: FieldsetRadio = {
+    options: [
+        {
+            dataTestId: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.SEND_ON_CTRL_ENTER}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.onForAllMessages'
+                    defaultMessage='On for all messages'
+                />
+            ),
+            name: `${AdvanceSettings.CTRL_SEND}`,
+            key: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.SEND_ON_CTRL_ENTER}`,
+            value: AdvanceSettings.SEND_ON_CTRL_ENTER,
+        },
+        {
+            dataTestId: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.CODE_BLOCK_CTRL_ENTER}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.onForCode'
+                    defaultMessage='On only for code blocks starting with ```'
+                />
+            ),
+            name: `${AdvanceSettings.CTRL_SEND}`,
+            key: `${AdvanceSettings.CTRL_SEND}-${AdvanceSettings.CODE_BLOCK_CTRL_ENTER}`,
+            value: AdvanceSettings.CODE_BLOCK_CTRL_ENTER,
+        },
+        {
+            dataTestId: `${AdvanceSettings.CTRL_SEND}-${OnAndOff.OFF}`,
+            title: (
+                <FormattedMessage
+                    id='user.settings.advance.off'
+                    defaultMessage='Off'
+                />
+            ),
+            name: `${AdvanceSettings.CTRL_SEND}`,
+            key: `${AdvanceSettings.CTRL_SEND}-${OnAndOff.OFF}`,
+            value: OnAndOff.OFF,
+        },
+    ],
+};
+
+export const FormattingInputFieldData: FieldsetRadio = {
+    options: [
+        {
+            dataTestId: `${AdvanceSettings.FORMATTING}-${OnAndOff.ON}`,
+            title: (
+                <FormattedMessage
+                    id='ser.settings.advance.on'
+                    defaultMessage='On'
+                />
+            ),
+            name: `${AdvanceSettings.FORMATTING}`,
+            key: `${AdvanceSettings.FORMATTING}-${OnAndOff.ON}`,
```

Comment:
Nitpicking, and I assume this is part of the existing component, but do we actually need separate `key` and `value` fields for each of these? A component's `key` only needs to be unique among that component's siblings, and I'd expect `value` to already be unique, so it doesn't seem necessary to have both